### PR TITLE
fix(GSX): Fixed GSX pin not actually disabling NWS

### LINKED ISSUE #7768: GSX Pushback Support
### Aircraft Version

Experimental

### Description

When doing pushbacks with GSX the NW STRG DISC message in the ECAM is not shown. Since recently we got full GSX support for Fuel and Payload in Experimental I would like to know if GSX is able to trigger the NW STRG DISC message on the FBW when inserting the bypass pin into the aircraft.

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

HanSolo80#3772

### PR DESCRIPTION
**Pull request by Maximilian-Reuter**
Fixes #7768 

## Summary of Changes

Does pretty much what is on the tin.

Nose wheel steering should now be disabled when the GSX bypass pin is inserted with the associated memo

## Additional context

Discord username (if different from GitHub):

## Testing instructions

1. Spawn C/D
2. Do normal setup up to push and start
3. Call GSX pushback
4. Observe ECAM 
5. NWS Disconnect message should show up when bypass pin is inserted
6. For completeness sake one should also do  a normal pushback using our integrated pushback

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Review by beheh — COMMENTED** [review]
I think this PR could be written better by splitting out a Bypass Pin struct from the PushbackTug concept. As I read it, the PushbackTug is really about the A32NX's own pushback tug, and what is checked by Hydraulics' NWS.

As it stands, `self.nose_wheel_steering_pin_inserted.output() || self.gsx_pin_inserted`
reads fairly weird now, because it's checking the timer for our own pushback and GSX, two somewhat unrelated concepts.

A better approach might be to split out the NWS pin as it's own struct, and then have that check both the A32NX pushback system (pushback.rs) and the GSX pin variable. That isolates our Pushback tug nicely and would also easily allow support for other addons in future.

File: fbw-common/src/wasm/systems/systems/src/hydraulic/bypass_pin.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    InitContext, Read, SimulationElement, SimulatorReader, SimulatorWriter, VariableIdentifier,
+    Write,
+};
+
+pub struct BypassPin {
+    gsx_pin_state_id: VariableIdentifier,
+    gsx_pin_inserted: bool,
+    nw_strg_disc_memo_id: VariableIdentifier,
+
+    bypass_pin_inserted: bool,
```
Comment (beheh) [inline fbw-common/src/wasm/systems/systems/src/hydraulic/bypass_pin.rs:12]: style: we tend to group VariableIdentifiers at the top, and then the rest of the state below.

```suggestion
    gsx_pin_state_id: VariableIdentifier,
    nw_strg_disc_memo_id: VariableIdentifier,

    gsx_pin_inserted: bool,
    bypass_pin_inserted: bool,
```

**Review by beheh — APPROVED** [review]

**Comment by Fragtality** [comment]
Made two quick Tests: works with Towbar and Towbarless! You can even now forget to switch on the Beacon *cough* 😅

**Comment by Saschl** [comment]
Quality Assurance Report

Discord : saschl
Object of testing: https://github.com/flybywiresim/aircraft/pull/8403
Tier of Testing : 1
Date : 13/1/2024

Testing Process:

Spawn at gate, power up plane and request pushback via GSX.
NWS ECAM memo appears after GSX pin is inserted.
PTU is inhibited when starting engines during pushback


Testing Results:
Passed

Conclusions:
ECAM memos and PTU Working as expected, normal Pushback also shows the same behaviour.

Media:

Screenshot 2024-01-13 002458
Screenshot 2024-01-13 002331

**Pull request merged**